src/App.js

```diff
@@ -147,12 +167,18 @@ function App() {
   );
 
   const message = useMemo(() => {
-    if (!guesses.length) return "Guess the five letter word";
-    if (guesses.length > 0 && !done) return "You're making progress!";
-    if (done && giveup === false) return `Congratulations! It took you ${guesses.length} guesses.`;
-    if (done && giveup === true) return `You gave up after ${guesses.length-1} guesses.`;
+    const randomMessage = (messages) => messages[Math.floor(Math.random() * messages.length)];
+    if (!guesses.length) return randomMessage(messages.welcome);
+    if (matches.newMatches === 1 && !done) return randomMessage(messages.newMatched1);
```

Comment:
A pattern that would make the code a little clearer and also run a bit more efficiently would be to:
```
if (!done) {
  if (matches.newMatches === 1) ...
  ...
  return randomMessage(messages.NoNew);
}
// at this point, you don't need to test for `done`
if (giveup === false) ...
...